### DISCUSSION TITLE
Configure for floor heating

### DISCUSSION THREAD

**Discussion by WMP** [jmcollin78/versatile_thermostat#221] [Q&A]
Hi, I have two floors in my house, on both of which I only have underfloor heating. At the moment I can only manually control the valves in the underfloor heating manifold, that is, I can open the flow completely, open some part of it, or close the flow completely if I don't want to heat that room. Thanks to the project https://github.com/fredlcore/BSB-LAN I can tell my boiler: "start heating and pump for underfloor heating on first floor", or "start heating and pump for underfloor heating on second floor".

However, I have a problem with overheating (when the temperature outside was around +5°C) or underheating (when the temperature outside was around -7°C): ![image](https://github.com/jmcollin78/versatile_thermostat/assets/45381/014ceb87-aed9-45a4-b394-655f46d7e9d6) ![image](https://github.com/jmcollin78/versatile_thermostat/assets/45381/3f3ec488-b67b-4c85-be1b-04f7cb82fe03)

My versatile settings are: ![image](https://github.com/jmcollin78/versatile_thermostat/assets/45381/b8dfc81f-c39a-4a22-8937-a948687ff0fd)

It seems to me that the overheating problem is caused by the inertia of the underfloor heating - the boiler heats until it reaches 22.5°C, then stops heating the water, but the hot water continues in the underfloor heating and gives off heat. However, I am unable to diagnose the problem of underheating water.

**Comment by jmcollin78** [comment]
You cycle is too long. 30 min in your case seems too long, your temperature curve seems to be relatively dynamic.

To better investigate I need curve like describe  in the readme (see at Apex-chart).

**Reply by WMP** [reply]
Thanks, I have changed the cycle length to 15 minutes. In the readme Apex-chart is set to `graph_span: 4h`. I think this is far too short: 

![image](https://github.com/jmcollin78/versatile_thermostat/assets/45381/2dc48c34-02ee-4b37-9f5e-3a5205425aba)

Here the view for 7 days (with group_by: func: avg duration: 4h):
![image](https://github.com/jmcollin78/versatile_thermostat/assets/45381/fa9b4586-7846-4aa8-acae-37dd497aaf8a)

**Comment by maia** [comment]
Do you define "overheating" as being 0.1°C-0.2°C above the target temperature? Because I'm not seeing more than that.

And early November 22 you can see that you increased the target temperature and with a delay of a few hours the room temperature rose – at a time where the target temperature was already decreased.

Keep in mind that you have a high latency, so increasing the target temperature *now* by 1°C means that the room temperature is 1°C higher *4-6 hours later*.

**Reply by WMP** [reply]
> Do you define "overheating" as being 0.1°C-0.2°C above the target temperature?

Yes, because supposedly 0.2°C, yet I feel it is too hot. Well, unless it's a problem with the loop in the living room, which is better heated due to the smaller window area to volume. I manually turned down the heating of this loop on the underfloor heating manifold.

> Keep in mind that you have a high latency, so increasing the target temperature now by 1°C means that the room temperature is 1°C higher 4-6 hours later.

Yes, I am aware of this, is there perhaps a setting for versatile which is more dedicated to underfloor heating?

**Comment by maia** [comment]
> Yes, because supposedly 0.2°C, yet I feel it is too hot.

Is it possible that the temperature sensor is in a place where it does not register changes in temperature quick enough, like far, far in a corner in 2.50m height? Or is it the floor that feels warm? (Which might indicate that the temperature flowing might be a bit too high)

> Yes, I am aware of this, is there perhaps a setting for versatile which is more dedicated to underfloor heating?

The "slow" setting in the adjusted regulation is the closest at the moment, but actually it's #208 that needs discussion and implementation to make some optimisations here. Besides that it's the human that needs to keep in mind that any change to the target temperature will need hours, so it's not helpful to have a schedule to turn the target temp up for an hour and then back again, because it will cause overheating a bit later.

**Comment by JoryHogeveen** [comment]
Just started VTherm and my situation is quite similar to the topic creator. Only floor heating thoughout the entire house and only manual valves. Everything is handled by a single climate entity (DIYLESS thermostat on Open Therm).

See graph:
![image](https://github.com/jmcollin78/versatile_thermostat/assets/826148/82e0541c-94b2-43e4-81f9-2f67de804de5)

The current temperature never went below the target temperature and still VTherm called for heat for 4 hours long.

Usually, once it reaches target temperature it will always overshoot around 0.3-0.5 degrees after 2 hours so if the thermostat calls for heat well past it already reached the target temperature it will overshoot a lot more.

The DIYLESS thermostat (in my case) works quite well but since this is house is insulated quite ok the cycle is very long and changing weather makes it nearly impossible for the PID algorithm to properly function as I would imagine (note: no expert here).

I'm using the "Slow" regulation as I would imagine (Like @maia said) this would be most suitable for floor heating with a regulation period of `10` minutes and a threshold of `0.2`.

Is this is too much of a different issue I'm happy to create a new topic :)
